osc tests fail with Python 3.15

Launching the tests of osc with python 3.15 fails (tested on current master 078f9110220599dfea209fa9c0759fd56c8a8cd2):
```
❯ python -m unittest discover
...................
The build root needs packages from project 'foo'.
Note that malicious packages can compromise the build result or even your system.
Well, goodbye then :-)
...Failed to load command class 'test.osc.commands.Child' because it references parent 'test.osc.commands.DoesNotExist' that doesn't exist
.....E............................................................../usr/lib64/python3.15/tempfile.py:484: ResourceWarning: Implicitly cleaning up <HTTPError 500: None>
  _warnings.warn(self.warn_message, ResourceWarning)
.......................................................................................................................................................................................................F...sssss....................................................................................................................................................................................EF............................................................................................./usr/lib64/python3.15/tempfile.py:484: ResourceWarning: Implicitly cleaning up <HTTPError 400: None>
  _warnings.warn(self.warn_message, ResourceWarning)
....................................................ssss.......................
======================================================================
ERROR: test_unknown_options (tests.test_commandline.TestCommandClasses.test_unknown_options)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/dan/suse/github.com/opensuse/osc/tests/test_commandline.py", line 126, in test_unknown_options
    self.assertRaises(SystemExit, main.parse_args, ["test-cmd", "--unknown-option"])
    ~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.15/unittest/case.py", line 835, in assertRaises
    return context.handle('assertRaises', args, kwargs)
           ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.15/unittest/case.py", line 245, in handle
    callable_obj(*args, **kwargs)
    ~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "/home/dan/suse/github.com/opensuse/osc/osc/commandline_common.py", line 295, in parse_args
    self.parser.error(f"unrecognized arguments: " + " ".join(unrecognized))
    ~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.15/argparse.py", line 2918, in error
    self.print_usage(_sys.stderr)
    ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "/usr/lib64/python3.15/argparse.py", line 2864, in print_usage
    formatter = self._get_formatter(file=file)
TypeError: OscArgumentParser._get_formatter() got an unexpected keyword argument 'file'

======================================================================
ERROR: testPrdiffDiffstat (tests.test_prdiff.TestProjectDiff.testPrdiffDiffstat)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib64/python3.15/unittest/mock.py", line 1439, in patched
    return func(*newargs, **newkeywargs)
  File "/home/dan/suse/github.com/opensuse/osc/tests/common.py", line 175, in wrapped_test_method
    test_method(self)
    ~~~~~~~~~~~^^^^^^
  File "/home/dan/suse/github.com/opensuse/osc/tests/common.py", line 175, in wrapped_test_method
    test_method(self)
    ~~~~~~~~~~~^^^^^^
  File "/usr/lib64/python3.15/unittest/mock.py", line 1439, in patched
    return func(*newargs, **newkeywargs)
  File "/home/dan/suse/github.com/opensuse/osc/tests/common.py", line 175, in wrapped_test_method
    test_method(self)
    ~~~~~~~~~~~^^^^^^
  File "/home/dan/suse/github.com/opensuse/osc/tests/common.py", line 175, in wrapped_test_method
    test_method(self)
    ~~~~~~~~~~~^^^^^^
  File "/usr/lib64/python3.15/unittest/mock.py", line 1439, in patched
    return func(*newargs, **newkeywargs)
  File "/home/dan/suse/github.com/opensuse/osc/tests/common.py", line 175, in wrapped_test_method
    test_method(self)
    ~~~~~~~~~~~^^^^^^
  File "/home/dan/suse/github.com/opensuse/osc/tests/test_prdiff.py", line 165, in testPrdiffDiffstat
    out = self._run_prdiff('--diffstat', 'old:prj', 'new:prj')
  File "/home/dan/suse/github.com/opensuse/osc/tests/test_prdiff.py", line 75, in _run_prdiff
    return self._run_osc("prdiff", *args)
           ~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "/home/dan/suse/github.com/opensuse/osc/tests/common.py", line 240, in _run_osc
    osc.commandline.OscMainCommand.main(argv)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^
  File "/home/dan/suse/github.com/opensuse/osc/osc/commandline.py", line 293, in main
    exit_code = cmd.run(args)
  File "/home/dan/suse/github.com/opensuse/osc/osc/commandline_common.py", line 221, in run
    return cmd.run(args)
           ~~~~~~~^^^^^^
  File "/home/dan/suse/github.com/opensuse/osc/osc/commandline.py", line 229, in run
    return self.func(args.command, args, *args.positional_args)
           ~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/dan/suse/github.com/opensuse/osc/osc/commandline.py", line 5335, in do_prdiff
    self._prdiff_output_diff(opts, rdiff)
    ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "/home/dan/suse/github.com/opensuse/osc/osc/commandline.py", line 5212, in _prdiff_output_diff
    with subprocess.Popen("diffstat", stdin=subprocess.PIPE, stdout=subprocess.PIPE, close_fds=True) as p:
         ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.15/subprocess.py", line 1207, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
    ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                        pass_fds, cwd, env,
                        ^^^^^^^^^^^^^^^^^^^
    ...<5 lines>...
                        gid, gids, uid, umask,
                        ^^^^^^^^^^^^^^^^^^^^^^
                        start_new_session, process_group)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.15/subprocess.py", line 2150, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'diffstat'

======================================================================
FAIL: test_plugin_locations (tests.test_doc_plugins.TestPopProjectPackageFromArgs.test_plugin_locations)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/dan/suse/github.com/opensuse/osc/tests/test_doc_plugins.py", line 79, in test_plugin_locations
    self.assertEqual(doc_paths, osc_paths)
    ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^
AssertionError: Lists differ: ['/usr/lib/osc-plugins', '/usr/local/lib/o[53 chars]ins'] != []

First list contains 4 additional elements.
First extra element 0:
'/usr/lib/osc-plugins'

+ []
- [
-     '/usr/lib/osc-plugins',
-     '/usr/local/lib/osc-plugins',
-     '~/.local/lib/osc-plugins',
-     '~/.osc-plugins',
- ]

======================================================================
FAIL: testPrdiffDiffstat (tests.test_prdiff.TestProjectDiff.testPrdiffDiffstat)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/dan/suse/github.com/opensuse/osc/tests/test_prdiff.py", line 65, in tearDown
    super().tearDown()
    ~~~~~~~~~~~~~~~~^^
  File "/home/dan/suse/github.com/opensuse/osc/tests/common.py", line 224, in tearDown
    self.assertTrue(len(EXPECTED_REQUESTS) == 0)
    ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: False is not true

----------------------------------------------------------------------
Ran 651 tests in 27.294s
```

--- a/osc/commandline_common.py
+++ b/osc/commandline_common.py
@@ -17,8 +17,10 @@ IN_VENV = getattr(sys, "real_prefix", sys.base_prefix) != sys.prefix
 
 
 class OscArgumentParser(argparse.ArgumentParser):
-    def _get_formatter(self):
-        # cache formatter to speed things a little bit up
+    def _get_formatter(self, *args, **kwargs):
+        # Accept extra parameters (like `file` in Python 3.15) while
+        # keeping the original behaviour: cache and return a formatter
+        # instance created with the parser program name.
         if not hasattr(self, "_formatter"):
             self._formatter = self.formatter_class(prog=self.prog)
         return self._formatter

